### PR TITLE
Bg fix add get sales

### PR DESCRIPTION
**Pull request by divinecharlotte**
### What does this PR do?

- [x] A Seller should be to get all his  sales and the admin get all sales for all sellers
- [x] fixing a bug from peer review on searching a product bt best before(0nce entered invalid date)
- [x] fixing the bug on login with google

### How should this be manually tested?

**Installation**
**1. Clone the repository by running** 
```
git clone https://github.com/atlp-rwanda/destructors-ec-bn.git
```
**2. Navigate to the project directory by running cd destructors-ec-bn.**
```
cd destructors-ec-bn
```
```
 cd  bg-fix-add-getSales
```
**3. Install the necessary dependencies by running npm install.**
```
npm install
```
**4. run migrations**
```
 npx sequelize-cli db:migrate
```
**5. run seeds**
```
npx sequelize-cli db:seed:all
```
**6. Run the development server by running npm run dev. This will start the server using Nodemon, which will automatically reload the server when changes are made to the code.**
```
npm run dev
```
**7. Run the tests by running npm run test** 
```
npm run test
```
### Any background context you want to provide?
when running this on a local server put into consideration the below information 👇 

- [x]  once you are a seller add the  seller `sign in  token` in headers with a key of `Authorization` and make sure that the value is started by `Bearer`
- [x]  once you are a admin add the  admin `sign in  token` in headers with a key of `Authorization` and make sure that the value is started by `Bearer`
- for finding sales  end point  **PATCH: /api/v1/sales**
- for searching products   end point  **PATCH: /api/v1/products/search**

### Questions:

- none

### CODE REVIEW

File: src/controllers/sale.controller.js [hunk 1]
```diff
@@ -84,4 +84,19 @@ const changeSaleStatus = async (req, res) => {
   
   }
 }
-export default changeSaleStatus;
\ No newline at end of file
+
+const getSales = async (req, res) => {
+
```
Comment (ba-lambert) [inline src/controllers/sale.controller.js:95]: Try to not use models inside controllers

File: src/routes/api/sale.routes.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
 import { Router } from 'express';
 import checkRole from '../../middlewares/checkRole';
-import changeSaleStatus from '../../controllers/sale.controller';
+import {changeSaleStatus, getSales } from '../../controllers/sale.controller';
 import extractToken from '../../middlewares/checkUserWithToken';
 const route = Router();
 
-
+route.get('/', extractToken, checkRole(['seller','admin']),getSales );
```
Comment (ba-lambert) [inline src/routes/api/sale.routes.js:8]: Also add password check middlware here

**Review by ba-lambert — CHANGES_REQUESTED** [review]

File: src/controllers/product.controller.js [hunk 3]
```diff
@@ -164,7 +164,14 @@ const searchProducts = async (req, res) => {
       if (minPrice) where.price = { [Op.gte]: minPrice };
       if (maxPrice) where.price = { ...where.price, [Op.lte]: maxPrice };
       if (categoryId) where.categoryId = categoryId;
```
Comment (katros1) [inline src/controllers/product.controller.js:179]: try to check also if the category id is uuid

File: src/controllers/product.controller.js [hunk 4]
```diff
@@ -156,22 +156,32 @@ const searchProducts = async (req, res) => {
...
+          return res.status(400).json({ error: 'Invalid date format for bestBefore' });
+        }
+        where.expiryDate = { [Op.lt]: expiryDate };
+      }
 
       if (UserData.role == 'seller') {
         where.sellerId = UserData.id;
       }
 
       const products = await Products.findAll({ where });
```
Comment (ba-lambert) [inline src/controllers/product.controller.js:193]: Try to not interact with database inside controllers

**Review by ba-lambert — DISMISSED** [review]

**Review by katros1 — APPROVED** [review]

**Comment by katros1** [comment]
Try to check your tests are failing!!!

File: src/controllers/sale.controller.js (hunk 1 above)
Comment (ba-lambert) [reply src/controllers/sale.controller.js:95]: You did not resolve the issue

Comment (ba-lambert) [reply src/controllers/sale.controller.js:95]: I mean that you have to respect the folder structuring

**Review by katros1 — APPROVED** [review]

**Review by buka1calvin — APPROVED** [review]

**Comment by buka1calvin** [comment]
divine please rebase

**Review by leandreAlly — APPROVED** [review]

**Review by mjeanbosco19 — APPROVED** [review]

**Review by JENNY-257 — APPROVED** [review]

**Review by Abdulkeza — APPROVED** [review]

**Pull request merged**